core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala

```diff
@@ -1001,32 +1041,220 @@ object TLVPoint extends Factory[TLVPoint] {
+      case 0 => false
+      case 1 => true
+      case b: Byte =>
+        throw new IllegalArgumentException(
+          s"Did not recognize leading byte: $b")
+    }
+
+    val withoutPrecision = BigSizeUInt(bytes.tail)
+    val extraPrecision = UInt16(
+      bytes.drop(1 + withoutPrecision.byteSize).take(2))
+
+    Signed16PTLVNumber(sign, withoutPrecision.toLong, extraPrecision.toInt)
+  }
+
+  def fromBigDecimal(number: BigDecimal): Signed16PTLVNumber = {
+    val sign = number >= 0
+    val withoutPrecision =
+      number.abs.setScale(0, RoundingMode.FLOOR).toLongExact
+    val extraPrecisionBD = (number.abs - withoutPrecision) * (1 << 16)
+    val extraPrecision =
+      extraPrecisionBD.setScale(0, RoundingMode.FLOOR).toIntExact
+
+    Signed16PTLVNumber(sign, withoutPrecision, extraPrecision)
+  }
+}
+
+case class HyperbolaPayoutCurvePieceTLV(
+    usePositivePiece: Boolean,
+    translateOutcome: Signed16PTLVNumber,
+    translatePayout: Signed16PTLVNumber,
+    a: Signed16PTLVNumber,
+    b: Signed16PTLVNumber,
+    c: Signed16PTLVNumber,
+    d: Signed16PTLVNumber)
+    extends PayoutCurvePieceTLV {
+  override val tpe: BigSizeUInt = HyperbolaPayoutCurvePieceTLV.tpe
+
+  override val value: ByteVector = {
+    boolBytes(usePositivePiece) ++
+      translateOutcome.bytes ++
+      translatePayout.bytes ++
+      a.bytes ++
+      b.bytes ++
+      c.bytes ++
+      d.bytes
   }
 }
 
+object HyperbolaPayoutCurvePieceTLV
+    extends TLVFactory[HyperbolaPayoutCurvePieceTLV] {
+  override val tpe: BigSizeUInt = BigSizeUInt(42794)
+
+  override def fromTLVValue(value: ByteVector): HyperbolaPayoutCurvePieceTLV = {
+    val iter = ValueIterator(value)
+
+    val usePositivePiece = iter.takeBoolean()
+    val translateOutcome = iter.take(Signed16PTLVNumber)
+    val translatePayout = iter.take(Signed16PTLVNumber)
+    val a = iter.take(Signed16PTLVNumber)
+    val b = iter.take(Signed16PTLVNumber)
+    val c = iter.take(Signed16PTLVNumber)
+    val d = iter.take(Signed16PTLVNumber)
+
+    HyperbolaPayoutCurvePieceTLV(usePositivePiece,
+                                 translateOutcome,
+                                 translatePayout,
+                                 a,
+                                 b,
+                                 c,
+                                 d)
+  }
+
+  override def typeName: String = "HyperbolaPayoutCurvePieceTLV"
+}
+
+case class OldPayoutFunctionV0TLV(points: Vector[OldTLVPoint])
+    extends DLCSetupPieceTLV {
+  override val tpe: BigSizeUInt = PayoutFunctionV0TLV.tpe
+
+  override val value: ByteVector = u16PrefixedList(points)
+}
+
 /** @see https://github.com/discreetlogcontracts/dlcspecs/blob/8ee4bbe816c9881c832b1ce320b9f14c72e3506f/NumericOutcome.md#curve-serialization */
-case class PayoutFunctionV0TLV(points: Vector[TLVPoint])
+case class PayoutFunctionV0TLV(
+    endpoints: Vector[TLVPoint],
+    pieces: Vector[PayoutCurvePieceTLV],
+    isOldSerialization: Boolean)
```

Comment:
kind of seems (based on the contents of this PR that this should maybe have default value `false`?

Reply:
I'd like to table this for now. 

When we get everything in #163 merged, i think its a great idea. For now, while development is still ongoing, i think it's less likely to lead to footguns if the developer has to explicitly state what serialization format they want